drivers/soundwire/sysfs_slave.c

```diff
@@ -212,3 +212,59 @@ int sdw_slave_sysfs_init(struct sdw_slave *slave)
 
 	return 0;
 }
+
+static ssize_t status_show(struct device *dev,
+			   struct device_attribute *attr, char *buf)
+{
+	struct sdw_slave *slave = dev_to_sdw_dev(dev);
+
+	switch (slave->status) {
+	case SDW_SLAVE_UNATTACHED:
+		return sprintf(buf, "%s", "UNATTACHED");
+	case SDW_SLAVE_ATTACHED:
+		return sprintf(buf, "%s", "ATTACHED");
+	case SDW_SLAVE_ALERT:
+		return sprintf(buf, "%s", "ALERT");
+	case SDW_SLAVE_RESERVED:
+		return sprintf(buf, "%s", "RESERVED");
+	}
+	return 0;
+}
+static DEVICE_ATTR_RO(status);
+
+static ssize_t device_number_show(struct device *dev,
+				  struct device_attribute *attr, char *buf)
+{
+	struct sdw_slave *slave = dev_to_sdw_dev(dev);
+
+	if (slave->status == SDW_SLAVE_UNATTACHED)
+		return sprintf(buf, "%s", "N/A");
+	else
+		return sprintf(buf, "%d", slave->dev_num);
```

Comment:
how about status == SDW_SLAVE_RESERVED ?

Reply:
the status is already shown in a separate entry, this is just the device_number.